app/src/app/components/contents/manageAccess/AddUserForm.tsx

```diff
@@ -0,0 +1,144 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Dispatch, SetStateAction, useState } from "react";
+import { useForm } from "react-hook-form";
+import { KeyedMutator } from "swr";
+import { z } from "zod";
+import appConfig from "../../../../config/appConfig";
+import { ApiError } from "../../../../lib/errors";
+import { fetcher } from "../../../../lib/fetch";
+import { Button } from "../../Base/Button";
+import { DialogClose, DialogFooter } from "../../Base/Dialog";
+import { Form, FormControl, FormDescription, FormField, FormItem, FormLabel, FormMessage } from "../../Base/Form";
+import { Input } from "../../Base/Input";
+import { RoleWithRelationships } from "./types/RoleWithRelationships";
+import {
+  MultiSelector,
+  MultiSelectorContent,
+  MultiSelectorInput,
+  MultiSelectorItem,
+  MultiSelectorList,
+  MultiSelectorTrigger
+} from "../../Base/MultiSelect";
+import { HttpStatus } from "../../../../lib/types/HttpStatus";
+
+interface AddUserFormProps {
+  mutate: KeyedMutator<RoleWithRelationships[]>;
+  setOpen: Dispatch<SetStateAction<boolean>>;
+  roleNames: string[];
+}
+export const AddUserForm = ({ mutate, setOpen, roleNames }: AddUserFormProps) => {
+  const [fetchError, setFetchError] = useState("");
+
+  const formSchema = z.object({
+    email: z.string().email(),
+    displayName: z.string().min(1),
+    password: z.string().min(8),
+    userRoles: z.array(z.string())
+  });
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      email: "",
+      displayName: "",
+      password: "",
+      userRoles: []
+    }
+  });
+  const onSubmit = async (values: z.infer<typeof formSchema>) => {
+    try {
+      await fetcher({
+        url: `${appConfig.apiUrl()}/user/basic`,
+        body: values,
+        method: "POST"
+      });
+      form.reset();
+      setOpen(false);
+      mutate();
+    } catch (error) {
+      console.error(error);
+      if (error instanceof ApiError && error.status === HttpStatus.BadRequest) {
+        return form.setError("email", { message: error.message });
```

Comment:
Is this assuming that if you get a BadRequest resons it's because something was wrong with the email address (presumably because it already exists?)

Reply:
yup its the only way to get bad request from this form